lib/auth/github.go

```diff
@@ -84,6 +91,118 @@ func (a *Server) upsertGithubConnector(ctx context.Context, connector types.Gith
 	return nil
 }
 
+// HTTPRequester allows a net/http.Client to be mocked for tests.
+type HTTPRequester interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// checkGithubOrgSSOSupport returns an error if any of the Github
+// organizations specified in this connector use external SSO. SSO
+// is a Teleport Enterprise feature that should not be allowed for OSS.
+// If userTeams is not nil, only organizations that are both specified
+// in conn and in userTeams will be checked. If client is nil a
+// net/http.Client will be used.
+func checkGithubOrgSSOSupport(ctx context.Context, conn types.GithubConnector, userTeams []teamResponse, orgCache *utils.FnCache, client HTTPRequester) error {
+	version := modules.GetModules().BuildType()
+	if version == modules.BuildEnterprise {
+		return nil
+	}
+
+	orgs := make(map[string]struct{})
+	addOrg := func(org string) {
+		if len(userTeams) != 0 {
+			// Only check organizations that the user is a member of and
+			// that are specified in this auth connector
+			for _, team := range userTeams {
+				if org == team.Org.Login {
+					orgs[org] = struct{}{}
+				}
+			}
+		} else {
+			orgs[org] = struct{}{}
+		}
+	}
+
+	// Check each organization only once
+	for _, mapping := range conn.GetTeamsToLogins() {
+		addOrg(mapping.Organization)
+	}
+	for _, mapping := range conn.GetTeamsToRoles() {
+		addOrg(mapping.Organization)
+	}
+
+	if client == nil {
+		transport, err := defaults.Transport()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		client = &http.Client{
+			Transport: transport,
+		}
```

Comment:
Nit: Since we have `defaults.Transport()`, would it make sense to make a `defaults.HTTPClient()` also that would return this? Would make this a little bit cleaner.

Reply:
Done